diff --git a/Src/moremeshsubs.F b/Src/moremeshsubs.F
index 506f8124d..6b8f63c0b 100644
--- a/Src/moremeshsubs.F
+++ b/Src/moremeshsubs.F
@@ -2749,7 +2749,11 @@
       integer           :: v
         !!
 
-      v = MERGE( ind, 0, present(ind) )
+      if ( present(ind) ) then
+         v = ind
+      else
+         v = 0
+      end if
       if (this%np==1) then
         v = v + 1
         Distr%box(:,:,v) = this%box
