diff --git a/CMakeLists.txt b/CMakeLists.txt
index 877616b70..ade6d879f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,6 +32,7 @@ include(GNUInstallDirs)
 
 # Include Siesta specific utility functions
 include(SiestaUtils)
+include(SiestaConda)
 siesta_util_ensure_out_of_source_build()
 
 if(PROJECT_IS_TOP_LEVEL)
diff --git a/Config/cmake/SiestaConda.cmake b/Config/cmake/SiestaConda.cmake
new file mode 100644
index 000000000..b4e3d19b2
--- /dev/null
+++ b/Config/cmake/SiestaConda.cmake
@@ -0,0 +1,46 @@
+# utility package to detect conda builds
+
+set(SIESTA_conda_build "$ENV{CONDA_BLD_PATH}")
+
+
+if(NOT SIESTA_conda_build)
+  message(STATUS "Not doing a conda-build")
+
+  macro(siesta_conda_fix var)
+  endmacro()
+
+  return()
+endif()
+
+list(APPEND CMAKE_MESSAGE_CONTEXT conda)
+message(STATUS "Building siesta in with conda")
+
+macro(siesta_conda_fix var)
+  string(REPLACE "$PREFIX"
+    "<prefix>"
+    ${var} "${${var}}"
+    )
+  string(REPLACE "$ENV{PREFIX}"
+    "<prefix>"
+    ${var} "${${var}}"
+    )
+  string(REPLACE "$BUILD_PREFIX"
+    "<prefix>/_build_env"
+    ${var} "${${var}}"
+    )
+  string(REPLACE "$ENV{BUILD_PREFIX}"
+    "<prefix>/_build_env"
+    ${var} "${${var}}"
+    )
+  string(REPLACE "$SRC_DIR"
+    "<prefix>/work"
+    ${var} "${${var}}"
+    )
+  string(REPLACE "$ENV{SRC_DIR}"
+    "<prefix>/work"
+    ${var} "${${var}}"
+    )
+endmacro()
+
+list(POP_BACK CMAKE_MESSAGE_INDENT)
+list(POP_BACK CMAKE_MESSAGE_CONTEXT)
diff --git a/Src/CMakeLists.txt b/Src/CMakeLists.txt
index 811be80bf..4635805fd 100644
--- a/Src/CMakeLists.txt
+++ b/Src/CMakeLists.txt
@@ -707,6 +707,7 @@ endif()
 # the maximum line length (132 characters) of Fortran 2003.
 # There is not check that the maximum number of continuation lines
 # (255 in Fortran 2003) is not exceeded.
+siesta_conda_fix(Fortran_FLAGS_CURRENT)
 siesta_get_multiline(LENGTH 128
   Fortran_FLAGS_CURRENT
   OUTPUT Fortran_FLAGS_CURRENT_multiline
@@ -721,6 +722,7 @@ get_target_property(
   COMPILE_DEFINITIONS
   )
 
+siesta_conda_fix(Fortran_PPFLAGS_CURRENT)
 siesta_get_multiline(LENGTH 128
   Fortran_PPFLAGS_CURRENT
   OUTPUT Fortran_PPFLAGS_CURRENT_multiline
diff --git a/Util/TS/PHtrans/CMakeLists.txt b/Util/TS/PHtrans/CMakeLists.txt
index c7ad33c75..55cfacdd7 100644
--- a/Util/TS/PHtrans/CMakeLists.txt
+++ b/Util/TS/PHtrans/CMakeLists.txt
@@ -211,6 +211,7 @@ target_compile_definitions(
 # the maximum line length (132 characters) of Fortran 2003.
 # There is not check that the maximum number of continuation lines
 # (255 in Fortran 2003) is not exceeded.
+siesta_conda_fix(Fortran_FLAGS_CURRENT)
 siesta_get_multiline(LENGTH 128
   Fortran_FLAGS_CURRENT
   OUTPUT Fortran_FLAGS_CURRENT_multiline
@@ -225,6 +226,7 @@ get_target_property(
   COMPILE_DEFINITIONS
   )
 
+siesta_conda_fix(Fortran_PPFLAGS_CURRENT)
 siesta_get_multiline(LENGTH 128
   Fortran_PPFLAGS_CURRENT
   OUTPUT Fortran_PPFLAGS_CURRENT_multiline
diff --git a/Util/TS/TBtrans/CMakeLists.txt b/Util/TS/TBtrans/CMakeLists.txt
index 3e4441873..5374ab0c7 100644
--- a/Util/TS/TBtrans/CMakeLists.txt
+++ b/Util/TS/TBtrans/CMakeLists.txt
@@ -205,6 +205,7 @@ target_compile_definitions(
 # the maximum line length (132 characters) of Fortran 2003.
 # There is not check that the maximum number of continuation lines
 # (255 in Fortran 2003) is not exceeded.
+siesta_conda_fix(Fortran_FLAGS_CURRENT)
 siesta_get_multiline(LENGTH 128
   Fortran_FLAGS_CURRENT
   OUTPUT Fortran_FLAGS_CURRENT_multiline
@@ -219,6 +220,7 @@ get_target_property(
   COMPILE_DEFINITIONS
   )
 
+siesta_conda_fix(Fortran_PPFLAGS_CURRENT)
 siesta_get_multiline(LENGTH 128
   Fortran_PPFLAGS_CURRENT
   OUTPUT Fortran_PPFLAGS_CURRENT_multiline
