diff --git a/CMakeLists.txt b/CMakeLists.txt
index 019699c1e..47bccc902 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -56,6 +56,7 @@ endif()
 include(SiestaFlags)
 include(SiestaCheckFlags)
 
+
 # -- Options ----------------------------------------
 
 # This is just to have the user choose how to do the build
@@ -757,10 +758,8 @@ siesta_print_feature_info(
     SIESTA_WITH_UNIT_CONVENTION
     SIESTA_WITH_MPI
     SIESTA_WITH_OPENMP
-    SIESTA_WITH_IPO
     SIESTA_WITH_NETCDF
     SIESTA_WITH_NCDF
-    SIESTA_WITH_NETCDF_PARALLEL
     SIESTA_WITH_LIBXC
     SIESTA_WITH_PEXSI
     SIESTA_WITH_FFTW
@@ -770,6 +769,11 @@ siesta_print_feature_info(
     SIESTA_WITH_ELSI
     SIESTA_WITH_ELPA
     SIESTA_WITH_FLOOK
+    # Compilation specific details
+    SIESTA_WITH_HOST_OPTIMIZATION
+    SIESTA_WITH_IPO
+    SIESTA_WITH_NETCDF_PARALLEL
+    SIESTA_WITH_NO_MPI_INTERFACES
     SIESTA_INSTALL
     BUILD_SHARED_LIBS
     SIESTA_SHARED_LIBS
@@ -777,7 +781,6 @@ siesta_print_feature_info(
     SIESTA_LINKER_FLAGS_POST
     SIESTA_LINKER_FLAGS
     SIESTA_TESTS
-    SIESTA_WITH_NO_MPI_INTERFACES
     SIESTA_TESTS_MPI_MIN_NUMPROCS
     SIESTA_TESTS_MPI_NUMPROCS
     SIESTA_TESTS_MPI_MAX_NUMPROCS
diff --git a/Changes/519.build.md b/Changes/519.build.md
new file mode 100644
index 000000000..d7f4982ae
--- /dev/null
+++ b/Changes/519.build.md
@@ -0,0 +1,6 @@
+Added build flag for removing host-specific
+compilation options.
+
+They are automatically removed when cross-compiling
+or when -DSIESTA_WITH_HOST_OPTIMIZATION=OFF.
+
diff --git a/INSTALL.md b/INSTALL.md
index 81e31afbb..342290564 100644
--- a/INSTALL.md
+++ b/INSTALL.md
@@ -653,6 +653,16 @@ for all options will be offered soon.
 
 ### Cross compiling
 
+Cross compilation is generally automatically detected and will remove
+any host-specific flags for the compilation. If one wishes to disable
+host-specific flags, one can use the flag:
+```
+   # When cmake does not detect cross-compilation
+   cmake ... -DSIESTA_WITH_HOST_OPTIMIZATION=OFF
+   # When cmake detects cross-compilation (i.e. -DCMAKE_SYSTEM_NAME=something)
+   cmake ... -DCMAKE_SYSTEM_NAME=alternate-system
+```
+
 When cross-compiling Siesta one needs to pass the appropriate kinds
 for the compiler used. This is necessary because the automatic
 generation of kind information would require the execution of
diff --git a/cmake/SiestaCheckFlags.cmake b/cmake/SiestaCheckFlags.cmake
index 30ad1521c..173db20e7 100644
--- a/cmake/SiestaCheckFlags.cmake
+++ b/cmake/SiestaCheckFlags.cmake
@@ -23,6 +23,25 @@ get_property(project_languages GLOBAL PROPERTY ENABLED_LANGUAGES)
 # Filter out the C, CXX and Fortran languages
 list(FILTER project_languages INCLUDE REGEX "^(CXX|C|Fortran)$")
 
+
+# When we are doing cross-compilation, we have to specify the exact run-time
+# flags, and we have to bypass the `try_run` runs by manually setting
+# the return values.
+# Manually specify flags for the try-runs, otherwise it'll fail
+# This should be called like this:
+#  siesta_try_run_default(run_test 0)
+#  try_run(run_test ...)
+# Which will then setup the run_test to a cache variable
+# when cross compiling.
+macro(siesta_try_run_default var)
+  if( ${ARGC} GREATER 0 )
+    set(${var} ${ARGV0} CACHE INTERNAL "Default value for the variable ${var}")
+  else()
+    set(${var} 0 CACHE INTERNAL "Default value for the variable ${var}")
+  endif()
+endmacro()
+
+
 set(compiler_ok TRUE)
 # Start checks
 message(CHECK_START "Checking default compiler usability and flags.")
@@ -45,6 +64,8 @@ foreach(lang IN LISTS project_languages)
 
   # check if a simple program can be compiled
   string(TOLOWER ${lang} lang_lower)
+  siesta_try_run_default(runs_${lang}_default TRUE)
+
   if(lang_lower MATCHES "fortran")
     check_fortran_source_runs([==[
 program t
diff --git a/cmake/SiestaCheckLinalg.cmake b/cmake/SiestaCheckLinalg.cmake
index 78d251697..114aa66d1 100644
--- a/cmake/SiestaCheckLinalg.cmake
+++ b/cmake/SiestaCheckLinalg.cmake
@@ -42,6 +42,7 @@ endif()
 #
 # Check that BLAS/LAPACK actually works
 #
+siesta_try_run_default(blas_sgemm_basic TRUE)
 check_fortran_source_runs(
 "
 real :: a(4), b(4), c(4)
@@ -69,6 +70,7 @@ endif()
 #
 # Now check the return-value convention
 #
+siesta_try_run_default(blas_cdotu_return_convention TRUE)
 check_fortran_source_runs(
 "
 complex :: c
diff --git a/cmake/SiestaFlags.cmake b/cmake/SiestaFlags.cmake
index 960bc60d2..9b0c421a1 100644
--- a/cmake/SiestaFlags.cmake
+++ b/cmake/SiestaFlags.cmake
@@ -24,6 +24,12 @@
 #
 include(CheckFortranCompilerFlag)
 
+
+option(SIESTA_WITH_HOST_OPTIMIZATION "Enable default host optimization flags" "ON")
+if( CMAKE_CROSSCOMPILING )
+  message(STATUS "Compiling with cross-compilation (CMAKE_CROSSCOMPILING)")
+endif()
+
 # Define linker flags used for the executables
 set(SIESTA_LINKER_FLAGS_PRE "" CACHE STRING
   "User defined linker flags prepended to the linker line")
@@ -53,10 +59,10 @@ endif()
 
 set(SIESTA_TOOLCHAIN "${_toolchain}"
   CACHE STRING
-  "Define which default compiler settings to use: none|generic|gnu|intel|mac|marconi100|marenostrum|vega"
+  "Define which default compiler settings to use: none|generic|gnu|intelllvm|intel|mac|marconi100|marenostrum|vega"
   )
 set_property(CACHE SIESTA_TOOLCHAIN PROPERTY STRINGS
-  "none" "generic" "gnu" "intel" "mac" "marconi100" "marenostrum" "vega")
+  "none" "generic" "gnu" "intelllvm" "intel" "mac" "marconi100" "marenostrum" "vega")
 
 foreach(_toolchain IN LISTS SIESTA_TOOLCHAIN)
   # Now load the toolchain
@@ -134,7 +140,7 @@ if(CMAKE_Fortran_COMPILER_ID MATCHES NVHPC)
   # an escape character (mostly for DFTD3).
   # The -Mnoinline flag is set only on files that need it, in the appropriate
   # CMakeLists.txt files
-  
+
   ## set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -Mbackslash -Mnoinline")
   set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -Mbackslash")
 endif()
diff --git a/cmake/SiestaPrintBuildInfo.cmake b/cmake/SiestaPrintBuildInfo.cmake
index 8ad33c7a8..84af263e7 100644
--- a/cmake/SiestaPrintBuildInfo.cmake
+++ b/cmake/SiestaPrintBuildInfo.cmake
@@ -451,6 +451,7 @@ siesta_print_feature_info(
   MSGON
     "How libraries and executables are linked"
   VARIABLES
+  BUILD_SHARED_LIBS
   CMAKE_EXE_LINKER_FLAGS_INIT
   CMAKE_EXE_LINKER_FLAGS
   CMAKE_BUILD_WITH_INSTALL_RPATH
@@ -464,6 +465,8 @@ siesta_print_feature_info(
   CMAKE_SHARED_LINKER_FLAGS
   CMAKE_STATIC_LINKER_FLAGS
   CMAKE_LINK_DEPENDS_NO_SHARED
+  CMAKE_SYSTEM_NAME
+  CMAKE_CROSSCOMPILING
 
   REQUIRED
   )
@@ -771,6 +774,8 @@ siesta_print_feature_info(
     "Handling the suffixes of the executables and libraries"
   VARIABLES
     SIESTA_SUFFIX
+    SIESTA_SET_RPATH
+    SIESTA_SHARED_LIBS
     SIESTA_EXECUTABLE_SUFFIX
     SIESTA_SHARED_LIBRARY_SUFFIX
     SIESTA_STATIC_LIBRARY_SUFFIX
@@ -780,8 +785,9 @@ siesta_print_feature_info(
 siesta_print_feature_info(
   NAME Kinds
   MSGON
-    "Cross-compiling and specification for kinds"
+    "Cross-compiling"
   VARIABLES
+    SIESTA_WITH_HOST_OPTIMIZATION
     SIESTA_INTEGER_KINDS
     SIESTA_REAL_KINDS
   REQUIRED
diff --git a/cmake/toolchains/cray.cmake b/cmake/toolchains/cray.cmake
index 356a17483..2c2a9b2d8 100644
--- a/cmake/toolchains/cray.cmake
+++ b/cmake/toolchains/cray.cmake
@@ -11,6 +11,10 @@
 #    manually
 #
 
+# Specific host variables
+# (Currently we don't have them for Cray)
+set(_host_flags "")
+
 
 #
 # Fortran compiler settings
@@ -18,15 +22,16 @@
 set(Fortran_FLAGS "${CMAKE_Fortran_FLAGS}"
   CACHE STRING "Build type independent Fortran compiler flags")
 
-set(Fortran_FLAGS_RELEASE ""
+# O3 is equivalent to fast
+set(Fortran_FLAGS_RELEASE "-O2 ${_host_flags}"
   CACHE STRING "Fortran compiler flags for Release build")
 
-set(Fortran_FLAGS_MINSIZEREL "-Os -march=native"
-  CACHE STRING "Fortran compiler flags for minimum size build")
-
 set(Fortran_FLAGS_RELWITHDEBINFO "-g ${Fortran_FLAGS_RELEASE}"
   CACHE STRING "Fortran compiler flags for Release with debug info build")
 
+set(Fortran_FLAGS_MINSIZEREL "${_host_flags}"
+  CACHE STRING "Fortran compiler flags for minimum size build")
+
 set(Fortran_FLAGS_DEBUG "-g -O0"
   CACHE STRING "Fortran compiler flags for Debug build")
 
@@ -40,12 +45,15 @@ set(Fortran_FLAGS_CHECK "-g -O0"
 set(C_FLAGS "${CMAKE_C_FLAGS}"
   CACHE STRING "Build type independent C compiler flags")
 
-set(C_FLAGS_RELEASE ""
+set(C_FLAGS_RELEASE "-O2 ${_host_flags}"
   CACHE STRING  "C compiler flags for Release build")
 
 set(C_FLAGS_RELWITDEBINFO "-g ${C_FLAGS_RELEASE}"
   CACHE STRING  "C compiler flags for RelWithDebInfo build")
 
+set(C_FLAGS_MINSIZEREL "${_host_flags}"
+  CACHE STRING "C compiler flags for minimum size build")
+
 set(C_FLAGS_DEBUG "-g -O0"
   CACHE STRING "C compiler flags for Debug build")
 
diff --git a/cmake/toolchains/gnu.cmake b/cmake/toolchains/gnu.cmake
index dc5993db1..3af63939d 100644
--- a/cmake/toolchains/gnu.cmake
+++ b/cmake/toolchains/gnu.cmake
@@ -17,6 +17,12 @@
 #    manually
 #
 
+# Specific host variables
+if( CMAKE_CROSSCOMPILING OR NOT SIESTA_WITH_HOST_OPTIMIZATION )
+  set(_host_flags "")
+else()
+  set(_host_flags "-march=native")
+endif()
 
 #
 # Fortran compiler settings
@@ -24,19 +30,19 @@
 set(Fortran_FLAGS "${CMAKE_Fortran_FLAGS}"
   CACHE STRING "Build type independent Fortran compiler flags")
 
-set(Fortran_FLAGS_RELEASE "-O3 -march=native"
+set(Fortran_FLAGS_RELEASE "-O3 ${_host_flags}"
   CACHE STRING "Fortran compiler flags for Release build")
 
-set(Fortran_FLAGS_MINSIZEREL "-Os -march=native"
-  CACHE STRING "Fortran compiler flags for minimum size build")
-
 set(Fortran_FLAGS_RELWITHDEBINFO "-g ${Fortran_FLAGS_RELEASE} -fbacktrace"
   CACHE STRING "Fortran compiler flags for Release with debug info build")
 
+set(Fortran_FLAGS_MINSIZEREL "-Os ${_host_flags}"
+  CACHE STRING "Fortran compiler flags for minimum size build")
+
 set(Fortran_FLAGS_DEBUG "-g -Og -fbacktrace"
   CACHE STRING "Fortran compiler flags for Debug build")
 
-set(Fortran_FLAGS_CHECK "-g -Og -fcheck=all -fbacktrace"
+set(Fortran_FLAGS_CHECK "-g -Og -fbacktrace -fcheck=all"
   CACHE STRING "Fortran compiler flags for Debug build")
 
 
@@ -46,12 +52,15 @@ set(Fortran_FLAGS_CHECK "-g -Og -fcheck=all -fbacktrace"
 set(C_FLAGS "${CMAKE_C_FLAGS}"
   CACHE STRING "Build type independent C compiler flags")
 
-set(C_FLAGS_RELEASE "-O3 -march=native"
+set(C_FLAGS_RELEASE "-O3 ${_host_flags}"
   CACHE STRING  "C compiler flags for Release build")
 
 set(C_FLAGS_RELWITDEBINFO "-g ${C_FLAGS_RELEASE}"
   CACHE STRING  "C compiler flags for RelWithDebInfo build")
 
+set(C_FLAGS_MINSIZEREL "-Os ${_host_flags}"
+  CACHE STRING "C compiler flags for minimum size build")
+
 set(C_FLAGS_DEBUG "-g -Wall -pedantic"
   CACHE STRING "C compiler flags for Debug build")
 
diff --git a/cmake/toolchains/intel.cmake b/cmake/toolchains/intel.cmake
index 53b6527ed..1bf9f4407 100644
--- a/cmake/toolchains/intel.cmake
+++ b/cmake/toolchains/intel.cmake
@@ -14,23 +14,29 @@
 #    manually
 #
 
+# Specific host variables
+if( CMAKE_CROSSCOMPILING OR NOT SIESTA_WITH_HOST_OPTIMIZATION )
+  set(_host_flags "")
+else()
+  set(_host_flags "-xHost")
+endif()
 
 #
 # Fortran compiler settings
 #
-set(Fortran_FLAGS_RELEASE "-O2 -ip -xHost -fp-model=strict -prec-div -prec-sqrt"
+set(Fortran_FLAGS_RELEASE "-O2 ${_host_flags} -ip -fp-model=strict -prec-div -prec-sqrt"
   CACHE STRING "Fortran compiler flags for Release build")
 
-set(Fortran_FLAGS_MINRELSIZE "-Os -ip -xHost -fp-model=strict -prec-div -prec-sqrt"
-    CACHE STRING "Fortran compiler flags for minimum size build")
-
 set(Fortran_FLAGS_RELWITHDEBINFO "-g ${Fortran_FLAGS_RELEASE}"
   CACHE STRING "Fortran compiler flags for Release with debug info build")
 
+set(Fortran_FLAGS_MINRELSIZE "-Os ${_host_flags} -ip -fp-model=strict -prec-div -prec-sqrt"
+    CACHE STRING "Fortran compiler flags for minimum size build")
+
 set(Fortran_FLAGS_DEBUG "-g -O0 -traceback"
   CACHE STRING "Fortran compiler flags for Debug build")
 
-set(Fortran_FLAGS_CHECK "-g -O0 -check all -traceback"
+set(Fortran_FLAGS_CHECK "-g -O0 -traceback -check all"
   CACHE STRING "Fortran compiler flags for Debug (checking) build")
 
 #
@@ -39,9 +45,15 @@ set(Fortran_FLAGS_CHECK "-g -O0 -check all -traceback"
 set(C_FLAGS "${CMAKE_C_FLAGS}"
   CACHE STRING "Build type independent C compiler flags")
 
-set(C_FLAGS_RELEASE "-O2 -ip -xHost"
+set(C_FLAGS_RELEASE "-O2 ${_host_flags} -ip"
   CACHE STRING  "C compiler flags for Release build")
 
+set(C_FLAGS_RELWITHDEBINFO "-g ${C_FLAGS_RELEASE}"
+  CACHE STRING  "C compiler flags for Release with debug info build")
+
+set(C_FLAGS_MINSIZEREL "-Os ${_host_flags} -ip"
+  CACHE STRING "C compiler flags for minimum size build")
+
 set(C_FLAGS_DEBUG "-g -O0 -traceback"
   CACHE STRING "C compiler flags for Debug build")
 
diff --git a/cmake/toolchains/intelllvm.cmake b/cmake/toolchains/intelllvm.cmake
index ffc34125d..c9991185d 100644
--- a/cmake/toolchains/intelllvm.cmake
+++ b/cmake/toolchains/intelllvm.cmake
@@ -14,23 +14,30 @@
 #    manually
 #
 
+# Specific host variables
+if( CMAKE_CROSSCOMPILING OR NOT SIESTA_WITH_HOST_OPTIMIZATION )
+  set(_host_flags "")
+else()
+  set(_host_flags "-xHost")
+endif()
+
 
 #
 # Fortran compiler settings
 #
-set(Fortran_FLAGS_RELEASE "-O2 -xHost -fp-model=strict"
+set(Fortran_FLAGS_RELEASE "-O2 ${_host_flags} -fp-model=strict"
   CACHE STRING "Fortran compiler flags for Release build")
 
-set(Fortran_FLAGS_MINRELSIZE "-Os -xHost -fp-model=strict"
-    CACHE STRING "Fortran compiler flags for minimum size build")
-
 set(Fortran_FLAGS_RELWITHDEBINFO "-g ${Fortran_FLAGS_RELEASE}"
   CACHE STRING "Fortran compiler flags for Release with debug info build")
 
+set(Fortran_FLAGS_MINRELSIZE "-Os ${_host_flags} -fp-model=strict"
+    CACHE STRING "Fortran compiler flags for minimum size build")
+
 set(Fortran_FLAGS_DEBUG "-g -O0 -traceback"
   CACHE STRING "Fortran compiler flags for Debug build")
 
-set(Fortran_FLAGS_CHECK "-g -O0 -check all -traceback"
+set(Fortran_FLAGS_CHECK "-g -O0 -traceback -check all"
   CACHE STRING "Fortran compiler flags for Debug (checking) build")
 
 #
@@ -39,9 +46,15 @@ set(Fortran_FLAGS_CHECK "-g -O0 -check all -traceback"
 set(C_FLAGS "${CMAKE_C_FLAGS}"
   CACHE STRING "Build type independent C compiler flags")
 
-set(C_FLAGS_RELEASE "-O2 -xHost"
+set(C_FLAGS_RELEASE "-O2 ${_host_flags}"
   CACHE STRING  "C compiler flags for Release build")
 
+set(C_FLAGS_RELWITHDEBINFO "-g ${C_FLAGS_RELEASE}"
+  CACHE STRING  "C compiler flags for Release with debug info build")
+
+set(C_FLAGS_MINSIZEREL "-Os ${_host_flags}"
+  CACHE STRING "C compiler flags for minimum size build")
+
 set(C_FLAGS_DEBUG "-g -O0"
   CACHE STRING "C compiler flags for Debug build")
 
diff --git a/cmake/toolchains/none.cmake b/cmake/toolchains/none.cmake
index 8474e086b..8ed72bf5b 100644
--- a/cmake/toolchains/none.cmake
+++ b/cmake/toolchains/none.cmake
@@ -51,6 +51,15 @@ set(C_FLAGS ""
 set(C_FLAGS_RELEASE ""
   CACHE STRING  "C compiler flags for Release build")
 
+set(C_FLAGS_RELWITHDEBINFO ""
+  CACHE STRING  "C compiler flags for Release with debug info build")
+
+set(C_FLAGS_MINSIZEREL ""
+  CACHE STRING "C compiler flags for minimum size build")
+
 set(C_FLAGS_DEBUG ""
   CACHE STRING "C compiler flags for Debug build")
 
+set(C_FLAGS_CHECK ""
+  CACHE STRING "C compiler flags for Check build")
+
diff --git a/cmake/toolchains/vega.cmake b/cmake/toolchains/vega.cmake
index 81fc57211..96ec2fff3 100644
--- a/cmake/toolchains/vega.cmake
+++ b/cmake/toolchains/vega.cmake
@@ -47,5 +47,5 @@ set(LAPACK_LIBRARY
 set(SCALAPACK_LIBRARY
      "-L/cvmfs/sling.si/modules/el7/software/ScaLAPACK/2.1.0-gompic-2020b/lib -lscalapack"
      CACHE STRING "scalapack library chosen")
-     
+
 set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE STRING "build_type")
