diff --git a/Src/Makefile b/Src/Makefile
index 2d25c6f25..bd58fa139 100644
--- a/Src/Makefile
+++ b/Src/Makefile
@@ -408,16 +408,19 @@ version.o: version
 version: version.F90
 	@echo
 	@echo "==> Incorporating information about present compilation (compiler and flags)"
-	@sed "s'SIESTA_ARCH'${SIESTA_ARCH}'g;s'FFLAGS'${FC} ${FFLAGS}'g;\
+	@sed "s'SIESTA_ARCH'${SIESTA_ARCH}'g;s'FFLAGS'$(shell basename ${FC}) ${FFLAGS}'g;\
 		s^COMPILER_VERSION^$(COMPILER_VERSION)^;\
 		s'SIESTA_VERSION'$$(cat ${<D}/../version.info)'g;\
 		s'LIBS'$(COMP_LIBS) $(LIBS)'g;\
-		s'FPPFLAGS'$(FPPFLAGS)'g" $<  > tmp.F90
+		s'-isystem [^[:space:]\"]*''g;\
+		s' -fdebug-prefix-map=[^[:space:]\"]*''g;\
+		s'FPPFLAGS'$(FPPFLAGS)  $(DEFS)'g" $<  > tmp.F90
 	@awk '{if (length>80) { cur=78; \
 		printf "%s&\n",substr($$0,0,78); \
 		while(length-cur>78){ cur=cur+76 ; \
 		printf "&%s&\n",substr($$0,cur-76,76) \
 		} printf "&%s\n",substr($$0,cur)} else { print $$0 }}' tmp.F90 > compinfo.F90
+	@cat compinfo.F90
 	$(MAKE) "FPPFLAGS=$(FPPFLAGS)" compinfo.o
 	@mv compinfo.o version.o
 	@echo
